packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -7,22 +7,41 @@
 import {ApiClients, Argument, DataType} from '../types'
 import {useAsync} from '../useAsync'
 import useCommerceApi from '../useCommerceApi'
-import {UseQueryResult} from '@tanstack/react-query'
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperProducts']
-
 /**
  * A hook for `ShopperProducts#getProducts`.
  * Allows access to multiple products by a single request. Only products that are online and assigned to a site catalog are returned. The maximum number of productIDs that can be requested are 24. Along with product details, the availability, images, price, promotions, and variations for the valid products will be included, as appropriate.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=getProducts} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperproducts.shopperproducts-1.html#getproducts} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProducts = (
-    arg: Argument<Client['getProducts']>
-): UseQueryResult<DataType<Client['getProducts']>, Error> => {
+type UseProductsParameters = NonNullable<Argument<Client['getProducts']>>['parameters']
+type UseProductsHeaders = NonNullable<Argument<Client['getProducts']>>['headers']
+type UseProductsArg = {headers?: UseProductsHeaders; rawResponse?: boolean} & UseProductsParameters
+function useProducts(
+    arg: Omit<UseProductsArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+): UseQueryResult<DataType<Client['getProducts']>, Error>
+function useProducts(
+    arg: Omit<UseProductsArg, 'rawResponse'> & {rawResponse: true},
+    options: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
```

Comment:
`options` is optional, right?

```suggestion
    options?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
```